### PR TITLE
[ISV-4941] Upgrade to Fedora 40 and Python 3.12

### PR DESCRIPTION
**Pull request by RichardPlesnik**

### CODE REVIEW

**Review by BorekZnovustvoritel — APPROVED** [review]
LGTM :+1:

File: pyproject.toml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tool.black]
-target_version = ['py311']
```
Comment (mporrato) [inline pyproject.toml:2]: Have you forgotten to refresh/commit the lock file? I would expect a change like this to cause a bunch of updates to the lock file.

Comment (mporrato) [reply pyproject.toml:2]: Just realized my comment wasn't clear enough: I wasn't referring to that specific line. Let me rephrase it:

Have you refreshed the lock file from a system with python 3.12?

File: operator-pipeline-images/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM registry.fedoraproject.org/fedora:38
+FROM images.paas.redhat.com/exd-sp-guild-isv/base-images:f40-latest
```
Comment (Allda) [inline operator-pipeline-images/Dockerfile:1]: This is an internal instance of the registry that is not available outside of the Red Hat VPN and you can't use it in public GitHub projects.

File: pyproject.toml (hunk 1 above)
Comment (RichardPlesnik) [reply pyproject.toml:2]: I tried to install it in python3.12 venv  both locally and in integration tests and it was running fine with the current lock file. But I can regenerate the new one if you want.

File: operator-pipeline-images/Dockerfile (hunk 2 above)
Comment (RichardPlesnik) [reply operator-pipeline-images/Dockerfile:1]: Thanks for pointing it out. I tried it without the kerberos authentication but apparently forgot to turn of my VPN before that. I will change it to the public fedora image.

**Review by mporrato — APPROVED** [review]

**Review by Allda — APPROVED** [review]
